runtime/runtime/src/lib.rs

```diff
@@ -946,6 +948,11 @@ impl Runtime {
         state_update.set(trie_key, global_contract_data.code.to_vec());
         state_update
             .commit(StateChangeCause::ReceiptProcessing { receipt_hash: receipt.get_hash() });
+        let _ = precompile_contract(
+            &ContractCode::new(global_contract_data.code.to_vec(), None),
```

Comment:
Try to avoid passing in `None` here if the hash is already available some other way (e.g. the global contract identifier is the hash.)

We have found in the past that computing hashes for contracts can be pretty expensive (though in this case it probably doesn't matter given how rarely the global contract deployments would happen.)

Reply:
understood, thanks, addressed in 6b58fed56